syntax = "proto3";

package shipping_notice;

import "google/protobuf/empty.proto";

service ShippingNoticeController {
    rpc List(ShippingNoticeListRequest) returns (stream ShippingNotice) {}
    rpc Create(ShippingNotice) returns (ShippingNotice) {}
    rpc Retrieve(ShippingNoticeRetrieveRequest) returns (ShippingNotice) {}
    rpc Update(ShippingNotice) returns (ShippingNotice) {}
    rpc Destroy(ShippingNotice) returns (google.protobuf.Empty) {}
}

message ShippingNotice {
    int64 id = 1;
    string code = 2;
    enum ShippingNoticeStatus {
        CREATED = 0;
        LOCKED = 1;
        SEALED = 2;
        SHIPPED = 3;
    }
    ShippingNoticeStatus status = 3;
    optional string template_url = 4;
    optional string locked_at = 5;
    optional string sealed_at = 6;
    optional string shipped_at = 7;
    string created_at = 8;
    string updated_at = 9;
}

message ShippingNoticeListRequest {
}

message ShippingNoticeRetrieveRequest {
    int64 id = 1;
}
